clippy_utils/src/check_proc_macro.rs

```diff
@@ -280,6 +298,11 @@ pub fn is_from_proc_macro<T: WithSearchPat>(cx: &T::Context, item: &T) -> bool {
     !span_matches_pat(cx.sess(), item.span(), start_pat, end_pat)
 }
 
+pub fn is_from_proc_macro_and_span(cx: &LateContext<'_>, item: &FnKind<'_>, span: Span) -> bool {
+    let (start_pat, end_pat) = fn_kind_pat(item);
+    !span_matches_pat(cx.sess(), span, start_pat, end_pat)
+}
```

Comment:
I wasn't sure how else to solve it. Unlike the other types this is implemented for, `FnKind` doesn't come with a span, so it has to be supplied by the caller manually.

Reply:
You can write an impl of `WithSearchPat` for `(FnKind, span)`.

Reply:
Done. I did a manual implementation because the tuple made it pretty weird, didn't want to ruin the macro just for that one implementation.

Reply:
I would have done the same.